src/data_vis.ts

```diff
@@ -193,44 +177,20 @@ class DataVisModel {
     };
 
     changeBurden(burden: string) {
-        this.humanReadableBurdenOutcome(burden)
+        this.humanReadableBurdenOutcome(burden);
         this.plotTitle(this.defaultTitle());
-        this.onUIChange(true, true);
     };
 
-    onUIChange(redraw: boolean, updateUI: boolean) {
-        const isTimeSeries: boolean = (this.currentPlot() == "Time series");
-        if (updateUI) {
-            // refilter the data
-            const filterOptions = {
-                metric: this.burdenOutcome(),
-                maxPlot: -1,
-                compare: this.compare(),
-                disagg: this.disaggregateBy(),
-                yearLow: this.yearFilter().selectedLow(),
-                yearHigh: this.yearFilter().selectedHigh(),
-                activityTypes: this.activityFilter().selectedOptions(),
-                selectedCountries: this.countryFilter().selectedOptions(),
-                selectedDiseases: this.diseaseFilter().selectedOptions(),
-                selectedVaccines: this.vaccineFilter().selectedOptions(),
-                selectedTouchstones: this.touchstoneFilter().selectedOptions(),
-                cumulative: (this.compare() == "year" && this.cumulativePlot()),
-                timeSeries: isTimeSeries
-            };
-
-            const filteredData = new DataFilterer().filterData(filterOptions, impactData, plotColours);
-            this.compareNames([...filteredData[1]]);
-        }
-
-        if (redraw) {
-            if (isTimeSeries) {
-                this.renderTimeSeries();
-            } else {
-                this.renderImpact();
-            }
-        }
-    };
+    updatePlotOptions() {
```

Comment:
So this `updatePlotOptions()` function replaces the `updateUI` bit of your `onUIChange` function.